src/components/sortingFunctions.js

```diff
@@ -0,0 +1,21 @@
+export const stylesFnx = (days) => {
+  if (days > 0 && days <= 7) {
+    return 'green';
+  } else if (days > 7 && days <= 30) {
+    return 'orange';
+  } else if (days > 30) {
+    return 'red';
+  } else {
+    return 'grey';
+  }
+};
+
```

Comment:
It is great you have created file and declared these functions in it.  I am not recommending you do it in this PR but I believe it would even be better to create a directory and name it `utils` where we declare utility functions like these. It will help tidy things up.

I have also noticed you are not checking whether an item is inactive as described in the last item state. I wonder why is that?

Reply:
It appears the `else` statement might not evaluate unless the value of `days` is less than `0` which I believe is not possible.

Reply:
Essentially let us start from the definition  of the item states to give you my understanding of the issue,
- Need to buy soon (fewer than 7 days)
- Need to buy kind of soon (between 7 & 30 days)
- Need to buy not soon (more than 30 days)
- Inactive (when there’s only 1 purchase in the database or the purchase is really out of date [the time that has elapsed since the last purchase is 2x what was estimated])

What makes the interpretation of this quite difficult and confusing, in my opinion (I could be wrong) is the fact that we are required to categorize the items into the different states  using three variables at the same time.
These variables are:
1. How soon one needs to buy the item or estimated number of days till next purchase
2. Number of purchases (Specifically for Inactive)
3. *Outdatedness *. (Specifically for Inactive)

From my interpretation, an Inactive item can also be in another category and vice versa.  Let us assume I bought an item the third time about 40 days ago and while I was making the purchase the app estimated my next purchase would be after 10 days. Right now the item  should be  in the `inactive` category because the number of actual days which have elapsed since I last purchased the item is 4 times the estimated days left for next purchase while buying the item. The item could also be in the `Need to buy kind of soon` category because of the estimated number of days. Which one of the two categorization variables, how soon one needs to buy the item and *outdatedness*, should take precedence when deciding in which category to put the item? Honestly I also don't know though I could go with *outdateness*.

In your code, I don't see you making reference to parameters for determining whether an item is inactive using the actual number of days which have elapsed versus estimated days left for next purchase. From the example I have given above, your code will mark the item which I last bought  approximately 40 days ago as `Need to buy kind of soon`  because it was estimated the item would be bought again after 10 days while making the last purchase. I believe it is correct to make that deduction. But the item could also be `Inactive` from the definition of an `Inactive` item given.

I am still not sure I understand what you mean by 
> we reasoned that days left till next purchase counts down towards 1. what happens when it counts to one and the user still hasn’t purchased the items again

In my understanding,  `daysLeftForNextPurchase` which you are passing to the function is only estimated during purchase and it remains constant between current and the next purchase.

I am really not sure about this as well. I could be over thinking things.

Reply:
ooh! let me introduce this discussion on slack we completely misunderstood it

Reply:
I totally agree that we making a utils directory and that we make our components code clean and more readable in the last two weeks by creating function files. 🙌🏽